view/frontend/web/js/amazonpay/button.js

```diff
@@ -0,0 +1,566 @@
+                let currency;
+
+                if (this.isProductView) {
+                    currency = currencyModel().getCurrency();
+                } else {
+                    const cartData =  customerData.get('cart');
+                    const adyenMethods = cartData()['adyen_payment_methods'];
+                    const paymentMethodExtraDetails = adyenMethods.paymentMethodsExtraDetails[amazonPaymentMethod.type];
+                    currency = paymentMethodExtraDetails.configuration.amount.currency;
+                };
+
+                url.searchParams.delete('amazonCheckoutSessionId');
+
+                const returnUrl = urlBuilder.build('checkout/cart/index');
+
+                return {
+                    showPayButton: true,
+                    productType: 'PayAndShip',
+                    currency: currency,
+                    environment: config.checkoutenv.toUpperCase(),
+                    returnUrl: returnUrl,
+                    configuration: {
+                        merchantId: amazonPaymentMethod.configuration.merchantId,
+                        publicKeyId: amazonPaymentMethod.configuration.publicKeyId,
+                        storeId: amazonPaymentMethod.configuration.storeId
+                    },
+                    onClick: function (resolve, reject) {validatePdpForm(resolve, reject, pdpForm);},
+                    onSubmit: function () {},
+                    onError: () => cancelCart(this.isProductView),
+                    ...amazonPayStyles
+                };
+            },
+
+            getAmazonPayOrderConfig: function (amazonPaymentMethod, element) {
+                const amazonPayStyles = getAmazonPayStyles();
+                const pdpForm = getPdpForm(element);
+                const amazonSessionKey = 'amazonCheckoutSessionId';
+                const url = new URL(location.href);
+                const amazonPaySessionKey = url.searchParams.get(amazonSessionKey);
+                let currency;
+
+                if (this.isProductView) {
+                    currency = currencyModel().getCurrency();
+                } else {
+                    const cartData =  customerData.get('cart');
+                    const adyenMethods = cartData()['adyen_payment_methods'];
+                    const paymentMethodExtraDetails = adyenMethods.paymentMethodsExtraDetails[amazonPaymentMethod.type];
+                    currency = paymentMethodExtraDetails.configuration.amount.currency;
+                };
+
+                url.searchParams.delete('amazonCheckoutSessionId');
+
+                const returnUrl = urlBuilder.build('checkout/cart/index' + '?amazonExpress=finalize');
+
+                return {
+                    amount: {
+                        value: this.isProductView
+                            ? formatAmount(totalsModel().getTotal() * 100)
+                            : formatAmount(window.checkoutConfig.quoteData.base_grand_total * 100),
```

Comment:
@peterojo here I changed how the amount is obtained if not product page

What still needs to be done is to refactor that part to not multiply by a 100, but rather create/use a utility class to sanitize and format it (to avoid issues with JPY)